data/trie/patriciaMerkleTrie.go

```diff
@@ -302,24 +302,47 @@ func (tr *patriciaMerkleTrie) Recreate(root []byte) (data.Trie, error) {
 		)
 	}
 
-	db := tr.trieStorage.GetDbThatContainsHash(root)
+	newTr := tr.recreateFromMainDb(root)
+	if newTr != nil {
+		return newTr, nil
+	}
+
+	return tr.recreateFromSnapshotDb(root)
+}
+
+func (tr *patriciaMerkleTrie) recreateFromMainDb(rootHash []byte) data.Trie {
+	_, err := tr.Database().Get(rootHash)
+	if err != nil {
+		return nil
+	}
+
+	newTr, _, err := tr.recreateFromDb(rootHash, tr.Database(), tr.trieStorage)
+	if err != nil {
+		log.Warn("trie recreate error:", "error", err, "root", hex.EncodeToString(rootHash))
```

Comment:
You can add return nil here, even if it works now, but if the code will expand below this line there could be a wrong return afterward.

Reply:
Done.